Core/Utilities/Plotting/Plots.py

```diff
@@ -32,64 +50,95 @@ def generateErrorMessagePlot( msgText ):
 
 
 def generateTimedStackedBarPlot( fileName, data, metadata ):
+  """
+  It is used to create a time based line plot.
+  :param str fileName the name of the file
+  :param list data the data which is used to create the plot
+  :param dict metadata extra information used to create the plot.
+  """
   try:
-    fn = file( fileName, "wb" )
-  except:
-    return S_ERROR( "Can't open %s" % fileName )
-  checkMetadata( metadata )
-  for key, value in ( ( 'sort_labels', 'sum' ), ( 'legend_unit', '%' ) ):
-    if key not in metadata:
-      metadata[ key ] = value
-  barGraph( data, fn, **metadata )
-  fn.close()
+    with open( fileName, "wb" ) as fn:
+      checkMetadata( metadata )
+      for key, value in ( ( 'sort_labels', 'sum' ), ( 'legend_unit', '%' ) ):
+        if key not in metadata:
+          metadata[ key ] = value
+      barGraph( data, fn, **metadata )
+      fn.close()
+  except IOError as e:
+    return S_ERROR( errno.EIO, e )
   return S_OK()
 
 def generateQualityPlot( fileName, data, metadata ):
+  """
+  It is used to create 2D plots.
+  
+  :param str fileName the name of the file
+  :param list data the data which is used to create the plot
+  :param dict metadata extra information used to create the plot.
+  """
   try:
-    fn = file( fileName, "wb" )
-  except:
-    return S_ERROR( "Can't open %s" % fileName )
-  checkMetadata( metadata )
-  metadata[ 'legend' ] = False
-  #HACK: Pad a bit to the left until the proper padding is calculated
-  maxKeyLength = max( [ len( key ) for key in data ] )
-  metadata[ 'sort_labels' ] = 'alpha'
-  metadata[ 'plot_left_padding' ] = int( maxKeyLength * 2.5 )
-  qualityGraph( data, fn, **metadata )
-  fn.close()
+    with open( fileName, "wb" ) as fn:
+      checkMetadata( metadata )
+      metadata[ 'legend' ] = False
+      #HACK: Pad a bit to the left until the proper padding is calculated
+      maxKeyLength = max( [ len( key ) for key in data ] )
+      metadata[ 'sort_labels' ] = 'alpha'
+      metadata[ 'plot_left_padding' ] = int( maxKeyLength * 2.5 )
+      qualityGraph( data, fn, **metadata )
+      fn.close()
```

Comment:
The good thing about using `with open()` is that you don't need to invoke the `close()` on the file descriptor: this is already done behind the scenes. It applies here and in other locations in this PR.